parquet/src/main/java/org/apache/iceberg/parquet/ParquetUtil.java

```diff
@@ -86,15 +95,23 @@ public static Metrics footerMetrics(ParquetMetadata metadata, MetricsConfig metr
     Map<Integer, Literal<?>> upperBounds = Maps.newHashMap();
     Set<Integer> missingStats = Sets.newHashSet();
 
-    MessageType parquetType = metadata.getFileMetaData().getSchema();
-    Schema fileSchema = ParquetSchemaUtil.convert(parquetType);
+    // ignore metrics for fields we failed to determine reliable IDs
+    MessageType parquetTypeWithIds = getParquetTypeWithIds(metadata, nameMapping);
+    Schema fileSchema = ParquetSchemaUtil.convert(parquetTypeWithIds, name -> null);
```

Comment:
What about naming this `convertAndPrune`?

Reply:
Updated.